breaking out event-codes-by-year into -count-non-stricken-event-codes

jtdevos opened the pull request:
…-by-year

jtdevos committed:
breaking out event-codes-by-year into -count-non-stricken-event-codes-by-year

jimlerza commented on scripts/reports/event-codes-by-year.ts:
![img](https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTqSJtqmxAwg0kbdvF-_1ZQ32x6FTAwV1l2-Q "Look how they massacred my boy")

jtdevos replied:
Yikes! Sorry I did not mean to commit that :-O

jimlerza commented on scripts/reports/count-non-stricken-event-codes-by-year.ts:
```diff
@@ -0,0 +1,134 @@
+  onlyNonStricken,
+  years,
+}: {
+  applicationContext: IApplicationContext;
```
We're supposed to use ServerApplicationContext now

jimlerza commented on scripts/reports/count-non-stricken-event-codes-by-year.ts:
You should remove "non-stricken" from the filename because that's optional.

jimlerza commented on scripts/reports/count-non-stricken-event-codes-by-year.ts:
```diff
@@ -0,0 +1,134 @@
+// usage: npx ts-node --transpile-only scripts/reports/count-non-stricken-event-codes-by-year.ts M071,M074 2021-2022 [includeStricken]  > ~/Desktop/m071s-and-m074s-filed-2021-2022.csv
```
Fix filename, also don't pipe the output to a CSV.

jtdevos replied:
Blargh - sorry, pipe removed.  What's wrong with the filename?

jimlerza replied:
See the previous comment - this report can optionally count stricken docket entries, so we don't need to specify "non-stricken" in the filename.

jimlerza commented on scripts/reports/count-non-stricken-event-codes-by-year.ts:
```diff
@@ -0,0 +1,134 @@
+requireEnvVars(['ENV', 'REGION']);
+
+const includeStricken = !!(process.argv.length > 4);
+if (includeStricken) {
+  console.log('including stricken docket entries');
+}
```
Is this still necessary? I thought the argument for not being verbose in the ultimate output was so this report could be used in other scripts.

jimlerza commented on scripts/reports/count-non-stricken-event-codes-by-year.ts:
```diff
@@ -0,0 +1,134 @@
+(async () => {
+  const applicationContext = createApplicationContext({});
+
+  const eventCodes = process.argv[2].split(',');
+  const years: number[] = [];
+  const yearArg = process.argv[3];
+  if (yearArg.includes('-')) {
+    const [lower, upper] = yearArg.split('-');
+    for (let i = Number(lower); i <= Number(upper); i++) {
+      years.push(Number(i));
+    }
+  } else {
+    const yearStrings = yearArg.split(',');
+    for (const year of yearStrings) {
+      years.push(Number(year));
+    }
+  }
```
Rather than proliferating copypasta, perhaps we can abstract this into a reports helper or something?

jtdevos committed:
reverting temporary changes I did not mean to commit

jtdevos committed:
adding requested changes (wip)

jimlerza committed:
Merge branch 'staging' into opex-count-event-codes

jtdevos committed:
incomplete work on extracting arg utils from eventcode reports (wip)

jtdevos committed:
cleanup args, usage (still WIP)

jtdevos committed:
more refactoring; added tests; report now executable (see notes)

- now using node:util:parseArgs instead of doing it manual
- pulled out parsing numbers, ranges from count-event-codes-by-year into a reportUtils file
- added exec shebang so count-event-codes can be run directly
- todo: similarly refactor event-codes-by-year

jimlerza committed:
Merge branch 'staging' into opex-count-event-codes

jimlerza committed:
Merge branch 'staging' into opex-count-event-codes

jimlerza committed:
Merge branch 'staging' into opex-count-event-codes

jimlerza committed:
Merge branch 'staging' into opex-count-event-codes

jtdevos committed:
refactor event-codes-by-year to use parseArgs; updated usage

jtdevos committed:
cleanup usage, set executable flag

jtdevos committed:
Merge branch 'staging' into opex-count-event-codes

jtdevos committed:
more usage cleanup

jtdevos committed:
Merge branch 'staging' into opex-count-event-codes

jimlerza committed:
Merge branch 'staging' into opex-count-event-codes

jimlerza committed:
Merge branch 'staging' into opex-count-event-codes

Mwindo commented on scripts/reports/count-event-codes-by-year.ts:
```diff
@@ -0,0 +1,170 @@
+}: {
+  applicationContext: ServerApplicationContext;
+  eventCodes: string[];
+  onlyNonStricken: boolean;
```
Any reason not to make this `includeStricken` as below?

Mwindo reviewed (approved):
Left a very minor comment, but LGTM. Thanks, @jtdevos!
